app/src/test/java/com/example/parkingsystem/ParkingReservationPresenterTest.java

```diff
@@ -0,0 +1,331 @@
+
+        // Then
+
+        Assert.assertTrue(startDate > nowDate);
+        Assert.assertTrue(endDate > nowDate);
+        Assert.assertTrue(startDate > endDate);
+    }
+
+    @Test
+    public void validateDates_bothDatesAreInThePast_isFalse() {
+        //Given
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, -3);
+        Date dateForStart = calendar.getTime();
+        long startDate = dateForStart.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, -2);
+        Date dateForEnd = calendar.getTime();
+        long endDate = dateForEnd.getTime();
+        long nowDate = new Date().getTime();
+
+        // When
+        presenter.validateDates(startDate, endDate);
+
+        // Then
+        Assert.assertTrue(startDate > endDate);
+        Assert.assertFalse(startDate > nowDate);
+        verify(view).showInconsistentDates();
+    }
+
+    @Test
+    public void validateDates_startDateIsInThePast_isFalse() {
+        //Given
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, -3);
+        Date dateForStart = calendar.getTime();
+        long startDate = dateForStart.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 5);
+        Date dateForEnd = calendar.getTime();
+        long endDate = dateForEnd.getTime();
+        long nowDate = new Date().getTime();
+
+        // When
+        presenter.validateDates(startDate, endDate);
+
+        // Then
+        Assert.assertTrue(startDate < endDate);
+        Assert.assertFalse(startDate > nowDate);
+        Assert.assertTrue(nowDate < endDate);
+        Assert.assertFalse(nowDate < startDate);
+
+        verify(view).showInconsistentDates();
+    }
+
+    @Test
+    public void validateDates_endDateIsInThePast_isFalse() {
+        //Given
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 3);
+        Date dateForStart = calendar.getTime();
+        long startDate = dateForStart.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, -5);
+        Date dateForEnd = calendar.getTime();
+        long endDate = dateForEnd.getTime();
+        long nowDate = new Date().getTime();
+
+        // When
+        presenter.validateDates(startDate, endDate);
+
+        // Then
+        Assert.assertTrue(startDate > endDate);
+        Assert.assertTrue(startDate > nowDate);
+        Assert.assertFalse(endDate > nowDate);
+
+        verify(view).showInconsistentDates();
+    }
+
+    @Test
+    public void validateDates_isNull() {
+        // When
+        presenter.validateDates(0, 0);
+        // Then
+        verify(view).showEmptyDates();
+    }
+
+    @Test
+    public void validateSecurityCode_isTrue() {
+        //Given
+        String code = "ADN123";
+        when(view.getSecurityCode()).thenReturn(code);
+
+        //When
+        presenter.validateSecurityCode(code);
+
+        //Then
+        Assert.assertNotNull(code);
```

Comment:
In this test you're only asserting "ADN123" != null. Instead, you should assert the correct result of  presenter.validateSecurityCode() and that view.showCodeNotComplaint() is never called.